src/lib/vocab/account/account.schema.ts

```diff
@@ -18,9 +21,24 @@ export const AccountModelSchema = {
 	properties: {
 		account_id: {type: 'number'},
 		name: {type: 'string'},
+		settings: {$ref: '/schemas/AccountSettings.json'},
 		created: {type: 'object', instanceof: 'Date', tsType: 'Date'},
 		updated: {anyOf: [{type: 'object', instanceof: 'Date', tsType: 'Date'}, {type: 'null'}]},
 	},
-	required: ['account_id', 'name', 'created', 'updated'],
+	required: ['account_id', 'name', 'settings', 'created', 'updated'],
+	additionalProperties: false,
+};
+
+export const AccountSettingsSchema = {
+	$id: '/schemas/AccountSettings.json',
+	type: 'object',
+	properties: {
+		darkmode: {type: 'boolean'},
+	},
+	required: ['darkmode'],
```

Comment:
```suggestion
	// required: [], // currently no settings are required, so we don't need migrations to add defaults for every property
```

I think this is better for now, maybe even as a permanent design choice. By making all settings optional, we won't have to do migrations for settings additions. Otherwise we'd have to change the migration above from `NOT NULL DEFAULT '{}';` to `NOT NULL DEFAULT '{"darkmode": false}';` and keep it in sync with `toDefaultAccountSettings `.